qir/stdlib/src/stdlib.c

```diff
@@ -96,7 +97,37 @@ int8_t *__quantum__rt__array_create_1d(int32_t size, int64_t n)
   *s           = n;
   *r           = 1;
   *a           = 0;
-  return ret;
+  return (struct Array *)ret;
+}
+
+int64_t __quantum__rt__array_get_size_1d(struct Array *array)
+{
+  int64_t *s = (int64_t *)((int8_t *)array + sizeof(int64_t));
+  return *s;
+}
+
+int8_t *__quantum__rt__array_get_element_ptr_1d(struct Array *array, int64_t n)
+{
+  int64_t *e = (int64_t *)(array);
+  return ((int8_t *)array + (*e) * n + QUANTUM_ARRRAY_RESERVED);
+}
+
+void __quantum__rt__array_update_alias_count(struct Array *arr, int32_t n)
+{
+  int64_t *a = (int64_t *)((int8_t *)arr + 3 * sizeof(int64_t));
+  *a += (int64_t)(n);
+}
+
+void __quantum__rt__array_update_reference_count(struct Array *arr, int32_t n)
+{
+  int64_t *r = (int64_t *)((int8_t *)arr + 2 * sizeof(int64_t));
+  *r += (int64_t)(n);
+  /*
+  if (*r <= 0)
+  {
+    free(arr);
+  }
+  */
```

Comment:
With this commented out, does it mean arrays will never be freed?

Reply:
This was a test to see if I could get things to perform better - resolving as it is not relevant for the latest version.